src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -443,22 +443,24 @@ static int fmi3_xml_value_boundary_check_strcmp(jm_string strVal, const char* fo
     return strcmp(strVal + idx_start, wbBuf) == 0 ? 0 : 1;
 }
 
-/* Convert str values of "false", "true", "0" and "1" to their corresponding boolean value
-   Values are read from input argument 'strVal' and the covnerted value put in 'field'.
-   Returns:
+/*
      0: OK
     -1: parsing failed
 */
-static int fmi3_xml_str_to_bool(fmi3_xml_parser_context_t *context, int required, void* field, void* defaultVal,
-        jm_string strVal, const fmi3_xml_primitive_type_t* primType) {
-    /* Should we do something with the values 'required' and 'defaultVal' or can we simply remove them? */
+static int fmi3_xml_str_to_bool(fmi3_xml_parser_context_t *context, int required, bool* field, bool* defaultVal,
+        jm_string strVal) {
     int status = -1;
+
+    if (!strVal && !required) {
+        field = defaultVal;
```

Comment:
*field

and shouldn't there be a case for !strVal && required? The strcmp below will fail, so will fail, but explicit check kind of makes sense. You can probably use same code as I added in #39 (just search for 'required').

Reply:
Or rather:
*field = *defaultVal

Or why does defaultVal have to be a pointer here?